conode/Makefile

```diff
@@ -1,54 +1,55 @@
 CONTAINER = conode
 IMAGE_NAME = c4dt/$(CONTAINER)
 TAG = dev-$(shell date +%y%m%d)
-WEBAPP_ASSET := ../webapp/src/assets/config.toml
+WEBAPP_ASSET := ../webapp/src/assets/config.local.toml
+BUILD_DIR := build/conodes
+BUILT := $(BUILD_DIR)/built
 
 all: docker
 
-.PHONY: cothority-pull conode newdb docker docker_run docker_wait docker_stop docker_start docker_clean clean
+.PHONY: newdb docker_run docker_wait docker_stop docker_start docker_clean clean
 
-.DEFAULT_GOAL := docker_run
+.DEFAULT_GOAL := $(BUILT)
 
-utils: cothority-pull
-	@echo "Compiling binaries"; \
-	go build -o build/conode; \
-	cd ../cothority && \
-		go build -o ../conode/build/bcadmin ./byzcoin/bcadmin && \
-		go build -o ../conode/build/csadmin ./calypso/csadmin && \
-		go build -o ../conode/build/phapp ./personhood/phapp
+$(BUILT): build/Dockerfile build/run_nodes.sh
+	make newdb
+	make docker
+	touch $(BUILT)
 
-cothority-pull:
-	make -C .. cothority
-
-newdb: utils
-	rm -f build/conodes/*db;
-	@echo "Starting nodes"; \
-	( cd build; COTHORITY_ALLOW_INSECURE_ADMIN=true ./run_nodes.sh -v 2 -t -n 4 -s -d conodes > /dev/null ) & \
-	make docker_wait; \
-	echo "Creating new ByzCoin and printing out important variables"; echo; \
-	rm -f *.cfg; \
-	( ./build/bcadmin -c . create build/conodes/public.toml --interval 500ms; \
-	  ./build/bcadmin latest --bc bc-*; \
-	  ./build/bcadmin key -print key-* ) | egrep "^(ByzCoin|Admin|Private|Creating Spawner)" | tee variables.txt; \
-	echo; echo "Shutting down conodes"; pkill -x conode; rm -f conodes/running
+newdb: docker
```

Comment:
hum, the "main" generated file in this section is "variables.txt". consider splitting it in two, one `newdb` which `rm build/conode` & `$(MAKE) variables.txt`, one `variables.txt` which does the main work.
it can then be depended upon in `$(WEBAPP_ASSET)`.

Reply:
In fact I couldn't get it to work nicely. Depending on `variables.txt` is a very nice idea, but then splitting it in two is difficult: `docker_start*` then cannot depend on `variables.txt`, but must depend on `newdb`, and so will create the docker everytime again.

But perhaps I'm missing something.